src/edit.js

```diff
@@ -1,19 +1,36 @@
-import countries from '../assets/countries.json';
-import './editor.scss';
+/**
+ * WordPress dependencies
+ */
 import { edit, globe } from '@wordpress/icons';
-import { BlockControls } from '@wordpress/block-editor';
-import { ComboboxControl, Placeholder, ToolbarButton, ToolbarGroup } from '@wordpress/components';
-import { getEmojiFlag } from './utils';
+import { BlockControls, useBlockProps } from '@wordpress/block-editor';
+import {
+	ComboboxControl,
+	Placeholder,
+	ToolbarButton,
+	ToolbarGroup,
+} from '@wordpress/components';
 import { useEffect, useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import countries from '../assets/countries.json';
+import { getEmojiFlag } from './utils';
 import Preview from './preview';
+import './editor.scss';
 
 export default function Edit( { attributes, setAttributes } ) {
 	const { countryCode, relatedPosts } = attributes;
-	const options = Object.keys(countries).map(code => ({
-        value: code,
-        label:  getEmojiFlag( code ) + '  ' + countries[code] + ' — ' + code
-	}));
+
+	const blockProps = useBlockProps( {
+		countryCode,
+		relatedPosts,
+	} );
+
+	const options = Object.keys( countries ).map( ( code ) => ( {
+		value: code,
+		label: getEmojiFlag( code ) + '  ' + countries[ code ] + ' — ' + code,
+	} ) );
```

Comment:
The `options` array does not depend on `Edit` props yet it's recalculated on every render. This should be moved outside of the `Edit` function.

Also, some nice to haves:
1. `Object.entries()` could be used to retrieve a tuple (key and value) at once.
2. JS template literal could be used for the label.

```suggestion
	const options = Object.entries( countries ).map( ( [ code, name ] ) => ( {
		value: code,
		label: `${ getEmojiFlag( code ) }  ${ name } — ${ code }`,
	} ) );
```